cmd/clusterctl/internal/util/objs.go

```diff
@@ -105,42 +120,87 @@ func FixImages(objs []unstructured.Unstructured, alterImageFunc func(image strin
 	// look for resources of kind Deployment and alter the image
 	for i := range objs {
```

Comment:
I cannot comment on 2 lines above here, but could we pass `objs` as `[]*unstructured.Unstructured` to remove the deref of pointers?

Reply:
What do you mean exactly?

Reply:
In the function definition, we're passing `objs` as a slice of non pointer unstructured objects, could we switch to use pointers instead?

Reply:
If I do this I'll brake the function interface and lot's of stuff that works with `[]unstructured.Unstructured` like this

https://github.com/kubernetes-sigs/cluster-api/blob/aaea64e524bfcc8d0cd429b2baa4e6956dd8d0bc/cmd/clusterctl/client/cluster/cert_manager.go#L353-L368

Reply:
I see, let's leave it for now then